addons/sale/views/res_partner_views.xml

```diff
@@ -43,7 +59,7 @@
         <field name="priority" eval="3"/>
         <field name="arch" type="xml">
             <div name="button_box" position="inside">
-                <button class="oe_stat_button" type="object" name="action_view_sale_order"
+                <button class="oe_stat_button" type="action" name="%(act_res_partner_2_sale_order_stat)d"
```

Comment:
In stable, the customers would have to update their modules for this to work. I think you'd need to keep the python method for now and delete it in master

Reply:
We will target master, not 18.0 ;)